### PR TITLE
Write gin-config to tensorboard

### PR DESCRIPTION
**Pull request by emailweixu**
So it's a little easier to compare experiments

### CODE REVIEW

File: alf/utils/common.py [hunk 1]
```diff
@@ -308,3 +309,79 @@ def reward_scaling(r, scale=1):
...
+    """Copy gin config files to root_dir
+
+    Args:
+        root_dir (str): directory path
+        gin_files (None|list[str]): list of file paths
+    """
+    root_dir = os.path.expanduser(root_dir)
+    os.makedirs(root_dir, exist_ok=True)
+    for f in gin_files:
+        shutil.copyfile(f, os.path.join(root_dir, os.path.basename(f)))
```
Comment (witwolf) [inline alf/utils/common.py]: import shutil this module

File: alf/utils/common.py [hunk 2]
```diff
@@ -308,3 +309,79 @@ def reward_scaling(r, scale=1):
...
+    
+    If FLAGS.gin_file is not set, find gin files under FLAGS.root_dir and
+    returns them.
+    Returns:
+        the gin file(s)
+    """
+    gin_file = flags.FLAGS.gin_file
+    if gin_file is None:
+        root_dir = os.path.expanduser(flags.FLAGS.root_dir)
+        gin_file = glob.glob(os.path.join(root_dir, "*.gin"))
```
Comment (witwolf) [inline alf/utils/common.py]: import glob and flags

File: alf/trainers/on_policy_trainer.py [hunk 3]
```diff
@@ -124,15 +126,19 @@ def train_():
...
+                        metrics=eval_metrics,
+                        environment=eval_env,
+                        state_spec=algorithm.predict_state_spec,
+                        action_fn=algorithm.greedy_predict,
+                        num_episodes=num_eval_episodes,
+                        train_step=global_step,
+                        summary_writer=eval_summary_writer,
+                        summary_prefix="Metrics")
                     metric_utils.log_metrics(eval_metrics)
+            if iter == 0:
```
Comment (witwolf) [inline alf/trainers/on_policy_trainer.py]: it's better to move  line 139-141   before the loop L144 ?

File: alf/utils/common.py (hunk 1 above)
Comment (emailweixu) [reply alf/utils/common.py]: It is imported in a different commit in my local repo. I'll change it.

File: alf/trainers/on_policy_trainer.py (hunk 3 above)
Comment (emailweixu) [reply alf/trainers/on_policy_trainer.py]: The functions need be called in order to get the operative configurations from gin. So we need to do the summary after the major functions

**Pull request merged**